internal/controller/ols_app_redis_reconciliator.go

```diff
@@ -78,3 +98,51 @@ func (r *OLSConfigReconciler) reconcileRedisService(ctx context.Context, cr *ols
 	r.logger.Info("OLS redis service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileRedisSecret(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	secret, err := r.generateRedisSecret(cr)
+	if err != nil {
+		return fmt.Errorf("failed to generate OLS redis secret: %w", err)
+	}
+	foundSecret := &corev1.Secret{}
+	err = r.Client.Get(ctx, client.ObjectKey{Name: cr.Spec.OLSConfig.ConversationCache.Redis.CredentialsSecretRef.Name, Namespace: cr.Namespace}, foundSecret)
+	if err != nil && errors.IsNotFound(err) {
+		labelSelector := labels.Set{"app.kubernetes.io/name": "lightspeed-service-redis"}.AsSelector()
+		oldSecrets := &corev1.SecretList{}
+		err = r.Client.List(ctx, oldSecrets, &client.ListOptions{Namespace: cr.Namespace, LabelSelector: labelSelector})
+		if err != nil {
+			return fmt.Errorf("failed to list old OLS redis secrets: %w", err)
+		}
+		for _, oldSecret := range oldSecrets.Items {
+			oldSecretCopy := oldSecret // Create a local copy of the loop variable to fix G601
+			if err := r.Client.Delete(ctx, &oldSecretCopy); err != nil {
+				return fmt.Errorf("failed to delete old OLS redis secret: %w", err)
+			}
+		}
+		err = r.Create(ctx, secret)
+		if err != nil {
+			return fmt.Errorf("failed to create OLS redis secret: %w", err)
+		}
+		r.stateCache[RedisSecretHashStateCacheKey] = secret.Annotations[RedisSecretHashKey]
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get OLS redis secret: %w", err)
+	}
+	foundSecretHash, err := hashBytes(foundSecret.Data[RedisSecretKeyName])
```

Comment:
@vishnuchalla Is this case to handle the password if any users directly changed in the Secret ? If the user updates the Secret directly we should revert the changes to original state.